src/lib.rs

```diff
@@ -43,13 +43,16 @@
 #![cfg_attr(feature="clippy", feature(plugin))]
 #![cfg_attr(feature="clippy", plugin(clippy))]
 
+#![allow(dead_code)]
+#![allow(unused_variables)]
```

Comment:
I'm in favor of not having it for he following reasons:

1. While developing, you normally change only a small portion of code. Here, unused warnings can be important to catch some bugs or keep track of what's pending/left to do.

2. It can catch new bugs, example if you forget to use a argument in a function etc.

3. We can disable it at a per-module basis (depending on where most of these warnings come from) instead of at the top-level.